browser/resources/settings/brave_clear_browsing_data_dialog/brave_clear_browsing_data_dialog_behavior.ts

```diff
@@ -1,151 +1,163 @@
   }
 
   override disconnectedCallback() {
     super.disconnectedCallback()
 
     if (this.onSelectedPageChangedCallback_) {
       this.$.pages.removeEventListener('selected-item-changed',
-        this.onSelectedPageChangedCallback_);
-      this.onSelectedPageChangedCallback_ = null;
+        this.onSelectedPageChangedCallback_)
+      this.onSelectedPageChangedCallback_ = null
     }
 
-    if (this.saveOnExitSettingsCallback_) {
-      this.shadowRoot.querySelector('#on-exit-tab').removeEventListener(
-        'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_);
-      this.updateSaveButtonStateCallback_ = null;
+    if (this.updateSaveButtonStateCallback_) {
+      this.shadowRoot!.querySelector('#on-exit-tab')!.removeEventListener(
+        'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_)
+      this.updateSaveButtonStateCallback_ = null
     }
 
     if (this.saveOnExitSettingsCallback_) {
-      this.shadowRoot.querySelector('#saveOnExitSettingsConfirm').removeEventListener(
-        'click', this.saveOnExitSettingsCallback_);
-      this.saveOnExitSettingsCallback_ = null;
+      this.shadowRoot!.
+        querySelector('#saveOnExitSettingsConfirm')!.
+        removeEventListener('click', this.saveOnExitSettingsCallback_)
+      this.saveOnExitSettingsCallback_ = null
     }
 
-    this.onClearBraveAdsDataClickHandler_ = () => {};
+    this.onClearBraveAdsDataClickHandler_ = () => {}
   }
 
 /**
   * Updates the text of a browsing data counter corresponding to the given
   * preference.
   * @param {string} prefName Browsing data type deletion preference.
   * @param {string} text The text with which to update the counter
-  * @private
   */
-  updateOnExitCountersText(prefName, text) {
+  private updateOnExitCountersText(prefName: string, text: string) {
     // Data type deletion preferences are named "browser.clear_data.<datatype>".
     // Strip the common prefix, i.e. use only "<datatype>".
-    const matches = prefName.match(/^browser\.clear_data\.(\w+)$/);
-    this.shadowRoot.querySelector('#on-exit-tab').setCounter(matches[1], text);
+    const matches = prefName.match(/^browser\.clear_data\.(\w+)$/)
+    if (matches) {
+      this.shadowRoot!.
+        querySelector<SettingsBraveClearBrowsingDataOnExitPageElement>(
+          '#on-exit-tab')!.setCounter(matches[1], text)
+    }
   }
 
   /**
    * Updates Clear and Save buttons visibility based on the selected tab.
-   * @private
    */
-  onSelectedPageChanged_() {
-    const page = this.$.pages.selectedItem;
+  private onSelectedPageChanged_() {
+    const page = this.$.pages.selectedItem
     if (!page) {
-      return;
+      return
     }
-    const isOnExitPage = (page.id === 'on-exit-tab');
-    this.$.clearButton.hidden = isOnExitPage;
-    this.shadowRoot.querySelector('#saveOnExitSettingsConfirm').hidden = !isOnExitPage;
+    const isOnExitPage = (page.id === 'on-exit-tab')
```

Comment:
optional (and unsure if linter approved) but I would've guessed:

```suggestion
    const isOnExitPage = page.id === 'on-exit-tab'
```

Reply:
Makes sense, fixed here e19393fcdc762a915a6e374b5ad4d9e39c8977b2